Doc/library/inspect.rst

```diff
@@ -1163,17 +1163,83 @@ Classes and functions
 The interpreter stack
 ---------------------
 
-When the following functions return "frame records," each record is a
-:term:`named tuple`
-``FrameInfo(frame, filename, lineno, function, code_context, index)``.
-The tuple contains the frame object, the filename, the line number of the
-current line,
-the function name, a list of lines of context from the source code, and the
-index of the current line within that list.
+When the following functions return ":class:`FrameInfo` objects," each record is a
+:class:`FrameInfo` object. For backwards compatibility these objects allow
+tuple-like operations on all attributes except ``positions``. This behavior
+is considered deprecated and may be removed in the future.
+
+.. class:: FrameInfo
+
+   .. attribute:: frame
+
+      The :ref:`frame object <frame-objects>` that the record corresponds to.
+
+   .. attribute:: filename
+
+      The file name associated with the code being executed by the frame this record
+      corresponds to.
+
+   .. attribute:: lineno
+
+      The line number of the current line associated with the code being
+      executed by the frame this record corresponds to.
+
+   .. attribute:: function
+
+      The function name that is being executed by the frame this record corresponds to.
+
+   .. attribute:: code_context
+
+      A list of lines of context from the source code that's being executed by the frame
+      this record corresponds to.
+
+   .. attribute:: index
+
+      The index of the current line being executed in the :attr:`code_context` list.
+
+   .. attribute:: positions
+
+      A :class:`dis.Positions` object containing the start line number, end line
+      number, start column offset and end column offset associated with the
+      instruction being executed by the frame this record corresponds to.
 
 .. versionchanged:: 3.5
    Return a named tuple instead of a tuple.
 
+.. versionchanged:: 3.11
+   Changed the return object from a namedtuple to a regular object (that is
+   backwards compatible with the previous named tuple).
+
+.. class:: Traceback
+
+   .. attribute:: filename
+
+      The file name associated with the code being executed by the frame this traceback
+      corresponds to.
+
+   .. attribute:: lineno
+
+      The line number of the current line associated with the code being
+      executed by the frame this traceback corresponds to.
+
+   .. attribute:: function
+
+      The function name that is being executed by the frame this traceback corresponds to.
+
+   .. attribute:: code_context
+
+      A list of lines of context from the source code that's being executed by the frame
+      this traceback corresponds to.
+
+   .. attribute:: index
+
+      The index of the current line being executed in the :attr:`code_context` list.
+
+   .. attribute:: positions
+
+      A tuple containing the start line number, end line number, start column offset and end column
```

Comment:
oxford comma before "and".  here and above.

As positions also relates to `:class:dis.Positions`... why not consistently return that type in this field? How about being consistent and always having this be a `dis.Positions` instance so that people can use its field names.

If we don't do that here for performance reasons or similar, specifically mention that this is always a plain tuple but is compatible with the field indices of dis.Positions. people who want named fields could use `dis.Positions(*tb.positions)` on it.

Reply:
> As positions also relates to :class:dis.Positions... why not consistently return that type in this field? How about being consistent and always having this be a dis.Positions instance so that people can use its field names.

I was already returning of of these objects but I didn't want to document that is exactly a `dis.Positions` in case we want to change it for something equivalent in the future in this particular module. But I assume it makes sense to just document it as users would be probably relying on that it anyway